### PR TITLE
Check if the directory exists before trying to create it

### LINKED ISSUE #6: mkdirp fails if the full destination directory already exists
Now I started using the new version and the behavior is still a bit odd and not exactly like `mkdir -p`.

If you run `mkdirp` on a full directory that already exists, it will throw a `550` error, because `(err.code === 550 && dirs.length > 0)` will be false, see https://github.com/sindresorhus/jsftp-mkdirp/blob/master/index.js#L28

I think we need multiple ftp commands here to properly map the expected `mkdir -p` behavior on ftp level.  The algorithm should look like this in pseudocode, just to give an idea:
```
INPUT: path

old_pwd = ftp.raw.pwd();

components = reverse(parentDirs(path));

Promise.mapSeries(components, function(component) {
    if (ftp.cwd(component) == error[550]) {
      return ftp.raw.mkd(component);
    } else {
      return true;
    }
  }).finally(function() {
    ftp.raw.cwd(old_pwd);
  })
```

So we basically just try to cd into each path component, if that fails at any point... run `ftp.raw.mkd` on that path component without _any_ error handling. Because if we cannot cd into an ftp directory and cannot create it, then everything is lost.

This will still do the expected thing wrt #4

### PR DESCRIPTION
**Pull request by ArtskydJ**
Supersedes #8. Fixes #6.

Checks if a directory exists. If it does, then it checks the next directory. If it doesn't, then it switches to creating directories instead of checking for their existence.

Also adds jsftp as a peer dep because it relies on the version of jsftp being at least v1.5.4.

/cc @hasufell

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,12 +40,15 @@
     "pify": "^2.3.0",
     "slash": "^1.0.0"
   },
+  "peerDependencies": {
+    "jsftp": "^1.5.4"
+  },
```
Comment (sindresorhus) [inline package.json]: Put it below `devDependencies`

Comment (ArtskydJ) [inline package.json]: Done

**Comment by ArtskydJ** [comment]
The build seems a bit flaky.  Now and then something throws with ECONNREFUSED. The tests are passing, aside from that.

File: package.json [hunk 2]
```diff
@@ -40,9 +40,6 @@
     "pify": "^2.3.0",
     "slash": "^1.0.0"
   },
-  "peerDependencies": {
-    "jsftp": "^1.5.4"
-  },
```
Comment (sindresorhus) [inline package.json]: That's not what I asked for.

File: index.js [hunk 3]
```diff
@@ -34,7 +34,22 @@ function mkdirp(dir) {
...
+	const checkIfDirExists = dir => {
+		dir = slash(dir);
+
+		return pify(this.raw.mlst.bind(this.raw))(dir)
+			.then(() => dirs.length > 0 && checkIfDirExists(dirs.pop()))
+			.catch(err => {
+				if (err.code !== 550 || dirs.length === 0) {
+					err.message += ` - mkd: ${dir}`;
+					throw err;
+				} else {
```
Comment (kevva) [inline index.js]: Unnecessary `else` statement since it's implicit that the code will stop executing if throwing.

Comment (ArtskydJ) [inline index.js]: I changed it to flow like mkdirp.

**Comment by sindresorhus** [comment]
Thanks @ArtskydJ :)

Gonna do a major bump since we now require a newer version as a peerDependency, just in case.

**Comment by ArtskydJ** [comment]
:grinning::+1:

**Pull request merged**